pkg/azuredisk/azuredisk.go

```diff
@@ -323,3 +338,48 @@ func IsCorruptedDir(dir string) bool {
 	fmt.Printf("IsCorruptedDir(%s) returned with error: %v", dir, pathErr)
 	return pathErr != nil && mount.IsCorruptedMnt(pathErr)
 }
+
+// setControllerCapabilities sets the controller capabilities field. It is intended for use with unit tests.
+func (d *DriverCore) setControllerCapabilities(caps []*csi.ControllerServiceCapability) {
```

Comment:
`setXXX` func is actually refactor, can we remove such code, by that way, the v2 implementation would be much simpler to review. And I think current operator assign is ok inside this driver, if there is requirement to set ControllerCapabilities outside of this package, then we need to write `SetControllerCapabilities` func. 
`setControllerCapabilities` func is only for internal use inside current package.

Reply:
The azuredisk unit tests currently have cases that call into internal methods. In order to allow the same unit tests to verify both the v1 & v2 driver, I needed to abstract them behind an interface which meant that I needed to add these accessor functions. (See FakeDriver in fake_azuredisk.go) Unfortunately as you note, this lead to more churn in this change than would otherwise be desirable. The churn in the unit test files should be limited to replacement of direct access to member with the use of these accessors.